jd4/case.py

```diff
@@ -256,20 +257,49 @@ def read_legacy_cases(config, open):
                           memory_bytes,
                           int(score_str))
 
-def read_yaml_cases(config, open):
-    for case in yaml.safe_load(config)['cases']:
-        if 'judge' not in case:
-            yield DefaultCase(partial(open, case['input']),
-                              partial(open, case['output']),
-                              parse_time_ns(case['time']),
-                              parse_memory_bytes(case['memory']),
-                              int(case['score']))
-        else:
-            yield CustomJudgeCase(partial(open, case['input']),
+def read_auto_cases(open, zip_file, time_limit='1s', memory_limit='128m'):
+    files = zip_file.namelist()
+    prefix = ''
+    cases = []
+    for i in files:
+        match = re_input.match(i)
+        if match:
+            prefix = match.group(1)
+            cases.append(int(match.group(2)))
+    if not len(cases):
+        raise FormatError('No test case found.')
+    cases.sort()
+    score = 100 // len(cases)
+    divider = len(cases) - 100 % len(cases)
```

Comment:
This code looks strange.

Reply:
But it actually works(

Reply:
Please explain your algorithm in comments.

Reply:
score for basic score
100%len(cases) is the extra score
and it should be sepeated.

e.g.
8 test cases:
100//len(cases)=12
100//len(cases)=4
len(cases)-4=4
which means test cases 0-3 will be 12/case
4-7 13/case

Reply:
@twd2 Resolve it if there's no problem.

Reply:
In comments in the code, not here.